### PR TITLE
REGR: re-add encoding for read_excel

### LINKED ISSUE #35753: BUG: read_excel not accepting encoding on 1.1.0
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
# Your code here
def date_parse(value: str) -> datetime.datetime:
    return datetime.datetime.strptime(value.strip(), "%d/%m/%y")

df = pd.read_excel(
    filepath,
    encoding="iso-8859-1",
    usecols="B",
    date_parser=date_parse,
    skiprows=4,
)
```

#### Problem description

TypeError: read_excel() got an unexpected keyword argument 'encoding'.

#### Expected Output

No error, as in previous versions.

#### Output of ``pd.show_versions()``

```INSTALLED VERSIONS
------------------
commit           : d9fff2792bf16178d4e450fe7384244e50635733
python           : 3.8.2.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-42-generic
Version          : #46-Ubuntu SMP Fri Jul 10 00:24:02 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : pt_BR.UTF-8

pandas           : 1.1.0
numpy            : 1.18.5
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.1
setuptools       : 41.2.0
Cython           : None
pytest           : 6.0.1
hypothesis       : None
sphinx           : 3.2.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
numba            : None```

</details>

### PR DESCRIPTION
**Pull request by twoertwein**
- [ ] closes #35753
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Needs tests!

### CODE REVIEW

File: doc/source/whatsnew/v1.1.1.rst [hunk 1]
```diff
@@ -27,6 +27,7 @@ Fixed regressions
 - Fixed regression where :meth:`DataFrame.merge_asof` would raise a ``UnboundLocalError`` when ``left_index`` , ``right_index`` and ``tolerance`` were set (:issue:`35558`)
 - Fixed regression in ``.groupby(..).rolling(..)`` where a custom ``BaseIndexer`` would be ignored (:issue:`35557`)
 - Fixed regression in :meth:`~pandas.core.groupby.DataFrameGroupBy.agg` where a list of functions would produce the wrong results if at least one of the functions did not aggregate. (:issue:`35490`)
+- Fixed regression in :func:`pandas.read_execl` where ``encoding`` was removed but is necessary for some files (:issue:`35753`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.1.1.rst:30]:
```suggestion
- Fixed regression in :func:`pandas.read_exec` where ``encoding`` was removed but is necessary for some files (:issue:`35753`)
```

**Comment by simonjayhawkins** [comment]
There were changes to `test_to_excel_output_encoding` in #34464, can that test be updated to cover this.

**Comment by WillAyd** [comment]
Does this do nothing or am I misreading it? Doesn't the excel format already define how to interpret its data?

**Review by jreback — CHANGES_REQUESTED** [review]
yeah we need a test here; I know you are trying to get one @twoertwein

**Comment by twoertwein** [comment]
> Does this do nothing or am I misreading it? Doesn't the excel format already define how to interpret its data?

I think it doesn't do what I expected it to do:

It will forward the `encoding` to https://github.com/pandas-dev/pandas/blob/13940c7f3c0371d6799bbd88b9c6546392b418a1/pandas/io/excel/_base.py#L494

but this is the wrong place (`encoding` finally ends up in `TextFileReader` but it seems to be never used there), it should (if we have an Excel file that needs a specific encoding) go to

https://github.com/pandas-dev/pandas/blob/13940c7f3c0371d6799bbd88b9c6546392b418a1/pandas/io/excel/_xlrd.py#L37

since this is the place (or one of the places) where the file is actually opened. After that, all functions seem to operate on strings so there shouldn't be a need for an encoding form there one.

I will put this PR in draft mode until we have an excel file that needs an encoding.

**Pull request closed without merging**